core/src/test/java/org/pentaho/platform/util/EmailerTest.java

```diff
@@ -82,4 +122,153 @@ public void setBcc_ReplaceToCommaTest() {
+  }
+
+  @Test
+  public void testEmbeddedHtmlBodyWithAttachment() throws IOException, MessagingException {
+    emailer.setTo( "tets@test.email.com" );
+    emailer.setFrom( "tets@test.email.com" );
+    emailer.setSubject( "Test" );
+    final String body = UUID.randomUUID().toString();
+    emailer.setBody( body );
+    emailer.setAttachmentMimeType( "text/html" );
+    try ( ByteArrayInputStream stream = new ByteArrayInputStream( eml.getBytes() ) ) {
+      emailer.setAttachment( stream );
+      emailer.setAttachmentName( "email-file" );
+    }
+    emailer.send();
+    assertEquals( 1, MockMail.size() );
+    final Message message = MockMail.get( 0 );
+    assertNotNull( message );
+    final MimeMultipart content = (MimeMultipart) message.getContent();
+    assertNotNull( content );
+    assertEquals( 2, content.getCount() );
+    final BodyPart bodyPart = content.getBodyPart( 0 );
+    assertNotNull( bodyPart );
+    assertEquals( "text/plain; charset=UTF-8", bodyPart.getContentType() );
+    final String bodyPartContent = (String) bodyPart.getContent();
+    assertEquals( body, bodyPartContent );
+  }
+
+  @Test
+  public void testGetSmtpSessionOAuth2() {
+    emailer.setSmtpHost( "smtp.outlook.com" );
+    emailer.setSmtpPort( 587 );
+    emailer.setTransportProtocol( "SMTP" );
+    emailer.setStartTLS( true );
+    emailer.setUseSSL( true );
+    emailer.setQuitWait( false );
+    emailer.setFromName( "test@pentaho.com" );
+    emailer.setUseAuthentication( true );
+    emailer.setAuthMechanism( "XOAUTH2" );
+    assertEquals( "587", emailer.getSmtpSession( true ).getProperty( "mail.smtp.port" ) );
+  }
+
+  @Test
+  public void testSend() {
+    emailer.setSmtpHost( "smtp.outlook.com" );
+    emailer.setSmtpPort( 587 );
+    emailer.setTransportProtocol( "SMTP" );
+    emailer.setStartTLS( true );
+    emailer.setUseSSL( true );
+    emailer.setQuitWait( false );
+    emailer.setFromName( "test@pentaho.com" );
+    emailer.setUseAuthentication( true );
+    emailer.setAuthMechanism( "basic" );
+    emailer.getSmtpSession( true );
+    emailer.setTo( "tets@test.email.com" );
+    emailer.setFrom( "tets@test.email.com" );
+    emailer.setSubject( "Test" );
+    final String body = UUID.randomUUID().toString();
+    emailer.setBody( body );
+    emailer.setAuthenticator( new Authenticator() {
+      protected PasswordAuthentication getPasswordAuthentication() {
+        String decrypted;
+        String pass = "cGFzc3dvcmQ=";
+        try {
+          Base64PasswordService ps = new Base64PasswordService();
+          if ( pass.startsWith( "ENC:" ) ) {
+            decrypted = ps.decrypt( pass.substring( 4, pass.length() ) );
+          } else {
+            decrypted = ps.decrypt( pass );
+          }
+        } catch ( Exception e ) {
+          decrypted = pass;
+        }
+        return new PasswordAuthentication( "test@pentaho.com", decrypted );
+      }
+    } );
+    assertEquals( true, emailer.send() );
```

Comment:
I don't see logic that verifies Emailer#send calls `service.sendEmail( emailConfig, token, session, msg );`